src/Microsoft.AspNetCore.Rewrite/Internal/Shared/PatternSegments/IsIPV6Segment.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Net.Sockets;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Rewrite.Internal.Shared.PatternSegments
+{
+
+    public class IsIPV6Segment : PatternSegment
+    {
+        public override string Evaluate(HttpContext context, MatchResults ruleMatch, MatchResults condMatch)
+        {
+            return context.Connection.LocalIpAddress.AddressFamily == AddressFamily.InterNetworkV6 ? "on" : "off";
```

Comment:
Use RemoteIpAddress instead. That will account for x-forwarded-for and reverse proxy scenarios. Also, the IP may be null